KeeTrayTOTP/Menu/MenuItemProviderBase.cs

```diff
@@ -0,0 +1,12 @@
+using System.Windows.Forms;
+
+namespace KeeTrayTOTP.Menu
+{
+    public abstract class MenuItemProviderBase
```

Comment:
This could probably be an interface. The Terminate method does nothing, so maybe let the MenuItem providers that need some shutdown logic check if that specific MenuItem provider implements IDisposable and dispose that over there?

The logic for disposing the menu items could then look something like:
```csharp
foreach (var disposable in new MenuItemProviderBase [] { TrayMenuItemProvider, EntryMenuItemProvider, MainMenuItemProvider }.OfType<IDisposable>())
{
     disposable.Dispose();
}
```

Reply:
Great idea, thanks!